# HG changeset patch
# User vhanquez@kneesa.uk.xensource.com
# Date Mon Apr 10 14:51:10 2006 +0000
# Node ID bdec77028194e6f51872b66b0575ed86ad542333
# parent: 07af1b34d4c49ac52faaff6cb315231655212316
fix bug #609 - error: PIC register `bx' clobbered in `asm'

Signed-off-by: Vincent Hanquez <vincent@xensource.com>


--- a/tools/ioemu/hw/vga.c	Mon Apr 10 14:47:51 2006 +0100
+++ b/tools/ioemu/hw/vga.c	Mon Apr 10 14:51:10 2006 +0000
@@ -1369,10 +1369,10 @@ static inline unsigned int cpuid_edx(uns
 {
     unsigned int eax, edx;
 
-    __asm__("cpuid"
+    __asm__("pushl %%ebx; cpuid; popl %%ebx"
             : "=a" (eax), "=d" (edx)
             : "0" (op)
-            : "bx", "cx");
+            : "cx");
 
     return edx;
 }

